src/Data/Poly/Internal/PolyOverFractional.hs

```diff
@@ -31,7 +32,7 @@ import qualified Data.Poly.Internal.Dense.Fractional as Dense (fractionalGcd)
 -- providing a faster 'GcdDomain' instance,
 -- when coefficients are 'Fractional'.
 newtype PolyOverFractional poly = PolyOverFractional { unPolyOverFractional :: poly }
-  deriving (Eq, Ord, Show, Num, Semiring, Semiring.Ring)
+  deriving (Eq, NFData, Num, Ord, Semiring.Ring, Semiring, Show)
```

Comment:
Is there a reason why Ring is imported qualified and Semiring is not?

Reply:
I have no idea - it was already there when I got here.